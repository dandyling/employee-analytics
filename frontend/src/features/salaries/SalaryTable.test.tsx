import React from "react";
import {
  calcTotalSalaries,
  getAllLocationRows,
  getSalaryStr,
  mapLocationRows,
  mapTableRows,
  SalaryTable,
} from "./SalaryTable";
import locationSalaries from "./../../stubs/LocationSalaries.json";
import locationRow from "./../../stubs/LocationRow.json";
import totalLocationRow from "./../../stubs/TotalLocationRow.json";

test("renders correctly", () => {
  const wrapper = <SalaryTable />;
  expect(wrapper).toMatchSnapshot();
});

test("map location rows correctly", () => {
  expect(mapLocationRows(locationSalaries)).toEqual(locationRow);
});

test("map empty location salaries correctly", () => {
  expect(mapLocationRows({})).toEqual([]);
});

test("calculate total correctly", () => {
  const total = {
    location: "Total",
    current: 52105.06184559034,
    previous: 52820.406048701305,
  };
  expect(calcTotalSalaries(locationRow)).toEqual(total);
});

test("map total rows correctly", () => {
  expect(getAllLocationRows(locationSalaries)).toEqual(totalLocationRow);
});

test("map total table rows correctly", () => {
  const locationRows = getAllLocationRows(locationSalaries);
  const wrapper = mapTableRows(locationRows);
  expect(wrapper).toMatchSnapshot();
});

test("get salary string format is correct", () => {
  expect(getSalaryStr(1000.23)).toEqual("$1,000");
});

test("get salary string format for 0 is correct", () => {
  expect(getSalaryStr(0)).toEqual("$0");
});

test("get salary string format for negative salary is correct", () => {
  expect(getSalaryStr(-1000.23)).toEqual("-$1,000");
});
