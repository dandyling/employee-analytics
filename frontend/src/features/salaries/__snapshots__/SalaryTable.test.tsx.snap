// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map total table rows correctly 1`] = `
Array [
  Object {
    "Delta": <DeltaChip
      current={8516.873377976195}
      previous={8446.234940476195}
    />,
    "Location": "China",
    "Salary": "$8,517",
  },
  Object {
    "Delta": <DeltaChip
      current={8305.026060606062}
      previous={8942.671060606059}
    />,
    "Location": "Japan",
    "Salary": "$8,305",
  },
  Object {
    "Delta": <DeltaChip
      current={8215.59226415095}
      previous={8651.396666666666}
    />,
    "Location": "Philippines",
    "Salary": "$8,216",
  },
  Object {
    "Delta": <DeltaChip
      current={8827.508833333333}
      previous={8552.770166666669}
    />,
    "Location": "Thailand",
    "Salary": "$8,828",
  },
  Object {
    "Delta": <DeltaChip
      current={9950.576666666664}
      previous={9080.97}
    />,
    "Location": "Malaysia",
    "Salary": "$9,951",
  },
  Object {
    "Delta": <DeltaChip
      current={8289.484642857142}
      previous={9146.363214285717}
    />,
    "Location": "Vietnam",
    "Salary": "$8,289",
  },
  Object {
    "Delta": <DeltaChip
      current={52105.06184559034}
      previous={52820.406048701305}
    />,
    "Location": "Total",
    "Salary": "$52,105",
  },
]
`;

exports[`renders correctly 1`] = `<SalaryTable />`;
